ambry-commons/src/main/java/com/github/ambry/compression/BaseCompression.java

```diff
@@ -222,6 +236,9 @@ public int compress(byte[] sourceBuffer, int sourceBufferOffset, int sourceDataS
     compressedBuffer[offset] = 1;
     offset++;
 
+    compressedBuffer[offset] = (byte) algorithmNameBinary.length;
+    offset++;
+
```

Comment:
why change any of this if it will be removed ?

Reply:
This change is necessary to make keep the current unit tests pass.  In previous implementation, the getAlgorithmNameBinary() returns both the size and the name.  I changed it to better match the method name to return just the name binary (no size).  That's why this change is necessary to keep existing unit tests pass.